src/smirnoff_hack.py

```diff
@@ -1,120 +1,158 @@
+        if cache_key not in AT_TOOLKIT_CACHE_assign_partial_charges:
+            at_original_assign_partial_charges(self, molecule, partial_charge_method=partial_charge_method, use_conformers=use_conformers, strict_n_conformers=strict_n_conformers, _cls=_cls)
+            AT_TOOLKIT_CACHE_assign_partial_charges[cache_key] = molecule.partial_charges
+        else:
+            molecule.partial_charges = AT_TOOLKIT_CACHE_assign_partial_charges[cache_key]
+        return
+    AmberToolsToolkitWrapper.assign_partial_charges = at_cached_assign_partial_charges
+
+
+    # cache the OE generate_conformers function (save 15s)
+    OE_TOOLKIT_CACHE_molecule_conformers = {}
+    oe_original_generate_conformers = OpenEyeToolkitWrapper.generate_conformers
+    def oe_cached_generate_conformers(self, molecule, n_conformers=1, rms_cutoff=None, clear_existing=True):
+        cache_key = hash((hash_molecule(molecule), n_conformers, str(rms_cutoff), clear_existing))
+        if cache_key not in OE_TOOLKIT_CACHE_molecule_conformers:
+            oe_original_generate_conformers(self, molecule, n_conformers=n_conformers, rms_cutoff=rms_cutoff, clear_existing=clear_existing)
+            OE_TOOLKIT_CACHE_molecule_conformers[cache_key] = molecule._conformers
+        molecule._conformers = OE_TOOLKIT_CACHE_molecule_conformers[cache_key]
+    OpenEyeToolkitWrapper.generate_conformers = oe_cached_generate_conformers
+
+
+    # cache the RDKit generate_conformers function
+    RDK_TOOLKIT_CACHE_molecule_conformers = {}
+    rdk_original_generate_conformers = RDKitToolkitWrapper.generate_conformers
+    def rdk_cached_generate_conformers(self, molecule, n_conformers=1, rms_cutoff=None, clear_existing=True):
+        cache_key = hash((hash_molecule(molecule), n_conformers, str(rms_cutoff), clear_existing))
+        if cache_key not in RDK_TOOLKIT_CACHE_molecule_conformers:
+            rdk_original_generate_conformers(self, molecule, n_conformers=n_conformers, rms_cutoff=rms_cutoff, clear_existing=clear_existing)
+            RDK_TOOLKIT_CACHE_molecule_conformers[cache_key] = molecule._conformers
+        molecule._conformers = RDK_TOOLKIT_CACHE_molecule_conformers[cache_key]
+    RDKitToolkitWrapper.generate_conformers = rdk_cached_generate_conformers
+
+
+    # final timing: 56s
+
+    # cache the ForceField creation (no longer needed since using OpenFF API for parameter modifications)
+
+    # import hashlib
+    # from openff.toolkit.typing.engines.smirnoff import ForceField
+    # SMIRNOFF_FORCE_FIELD_CACHE = {}
+    # def getForceField(*ffpaths):
+    #     hasher = hashlib.md5()
+    #     for path in ffpaths:
+    #         with open(path, 'rb') as f:
+    #             hasher.update(f.read())
+    #     cache_key = hasher.hexdigest()
+    #     if cache_key not in SMIRNOFF_FORCE_FIELD_CACHE:
+    #         SMIRNOFF_FORCE_FIELD_CACHE[cache_key] = ForceField(*ffpaths, allow_cosmetic_attributes=True)
+    #     return SMIRNOFF_FORCE_FIELD_CACHE[cache_key]
```

Comment:
None of this code has changed - it has just been indented by 4 spaces to fall under the `if` statement.